### PR TITLE
Improve logging for failed JSON serialization

### LINKED ISSUE #1218: Improve logging for failed JSON serialization
Not helpful to pin point the issue:

The issue happens consistently on the Unity SDK with the `ThrowKotlinBackground` sample:

```
09-28 19:16:52.779 16991 17026 I Unity   : Sentry: (Debug) Initializing Hub for Dsn: 'https://94677106febe46b88b9b9ae5efd18a00@o447951.ingest.sentry.io/5439417'.
09-28 19:16:52.779 16991 17026 I Unity   : Sentry: (Debug) Using no request body compression strategy.
09-28 19:16:52.788 16991 17026 I Unity   : Sentry: (Debug) Flushing existing cache during transport activation up to 00:00:02.
09-28 19:16:52.789 16991 17026 I Unity   : Sentry: (Debug) External FlushAsync invocation: flushing cached envelopes.
09-28 19:16:52.792 16991 17026 I Unity   : Sentry: (Debug) No cached file to process.
09-28 19:16:52.795 16991 17070 I Unity   : Sentry: (Debug) Moving unprocessed file back to cache: /storage/emulated/0/Android/data/io.sentry.samples.unityofbugs/files/Sentry/618762E37120B6E01219C43B9076FFEC1ACE8406/__processing/1632865550_1059__-372960834.envelope to /storage/emulated/0/Android/data/io.sentry.samples.unityofbugs/files/Sentry/618762E37120B6E01219C43B9076FFEC1ACE8406/1632865550_1059__-372960834.envelope.
09-28 19:16:52.795 16991 17026 I Unity   : Sentry: (Debug) Registering integration: 'AutoSessionTrackingIntegration'.
09-28 19:16:52.795 16991 17026 I Unity   : Sentry: (Debug) Attempting to recover persisted session from file.
09-28 19:16:52.796 16991 17070 I Unity   : Sentry: (Debug) Worker signal triggered: flushing cached envelopes.
09-28 19:16:52.799 16991 17070 I Unity   : Sentry: (Debug) Reading cached envelope: /storage/emulated/0/Android/data/io.sentry.samples.unityofbugs/files/Sentry/618762E37120B6E01219C43B9076FFEC1ACE8406/__processing/1632865550_1059__-372960834.envelope
09-28 19:16:52.816 16991 17026 I Unity   : Sentry: (Debug) Native Android SDK reported: 'crashedLastRun': 'True'
09-28 19:16:52.817 16991 17026 I Unity   : Sentry: (Info) Recovered session: EndStatus: Crashed
09-28 19:16:52.821 16991 17026 I Unity   : Sentry: (Info) Envelope queued up: ''
09-28 19:16:52.822 16991 17072 I Unity   : Sentry: (Debug) Storing file /storage/emulated/0/Android/data/io.sentry.samples.unityofbugs/files/Sentry/618762E37120B6E01219C43B9076FFEC1ACE8406/1632871012_6820__1029138982.envelope.
09-28 19:16:52.822 16991 17026 I Unity   : Sentry: (Debug) Created directory for installation ID file (/storage/emulated/0/Android/data/io.sentry.samples.unityofbugs/files/Sentry/618762E37120B6E01219C43B9076FFEC1ACE8406).
09-28 19:16:52.823 16991 17026 I Unity   : Sentry: (Debug) Resolved installation ID '474bc867-1fe0-492f-8ee7-0eaaec51dafd'.
09-28 19:16:52.826 16991 17072 I Unity   : Sentry: (Debug) Envelope  handed off to transport. #1 in queue.
09-28 19:16:52.844 16991 17070 E Unity   : Sentry: (Error) Exception in background worker of CachingTransport. System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
09-28 19:16:52.844 16991 17070 E Unity   :   at System.Text.Json.ThrowHelper.ThrowJsonReaderException (System.Text.Json.Utf8JsonReader& json, System.Text.Json.ExceptionResource resource, System.Byte nextByte, System.ReadOnlySpan`1[T] bytes) [0x00000] in <00000000000000000000000000000000>:0
09-28 19:16:52.844 16991 17070 E Unity   :   at System.Text.Json.Utf8JsonReader.Read () [0x00000] in <00000000000000000000000000000000>:0
09-28 19:16:52.844 16991 17070 E Unity   :   at System.Text.Json.JsonDocument.Parse (System.ReadOnlySpan`1[T] utf8JsonSpan, System.Text.Json.JsonReaderOptions readerOptions, System.Text.Json.JsonDocument+MetadataDb& database, System.Text.Json.JsonDocument+StackRowStack& stack) [0x00000] in <00000000000000000000000000000000>:0
09-28 19:16:52.844 16991 17070 E Unity   :   at System.Text.Json.JsonDocument.Parse (System.ReadOnlyMemory`1[T] utf8Json, System.Text.Json.JsonReaderOptions readerOpt
09-28 19:16:53.333 16991 17026 I Unity   : Sentry: (Info) Started new session (SID: f260386d8b9042c68221da9c4ce15842; DID: 474bc867-1fe0-492f-8ee7-0eaaec51dafd).
09-28 19:16:53.333 16991 17026 I Unity   : Sentry: (Debug) Persisting session (SID: 'f260386d8b9042c68221da9c4ce15842') to a file.
09-28 19:16:53.334 16991 17026 I Unity   : Sentry: (Debug) Created persistence directory for session file '/storage/emulated/0/Android/data/io.sentry.samples.unityofbugs/files/Sentry/618762E37120B6E01219C43B9076FFEC1ACE8406'.
09-28 19:16:53.335 16991 17026 I Unity   : Sentry: (Debug) Persisted session to a file '/storage/emulated/0/Android/data/io.sentry.samples.unityofbugs/files/Sentry/618762E37120B6E01219C43B9076FFEC1ACE8406/.session'.
09-28 19:16:53.336 16991 17026 I Unity   : Sentry: (Info) Envelope queued up: ''
09-28 19:16:53.336 16991 17026 I Unity   : Sentry: (Debug) Registering integration: 'AppDomainUnhandledExceptionIntegration'.
09-28 19:16:53.336 16991 17026 I Unity   : Sentry: (Debug) Registering integration: 'AppDomainProcessExitIntegration'.
09-28 19:16:53.336 16991 17026 I Unity   : Sentry: (Debug) Registering integration: 'TaskUnobservedTaskExceptionIntegration'.
09-28 19:16:53.336 16991 17026 I Unity   : Sentry: (Debug) Registering integration: 'UnityApplicationLoggingIntegration'.
09-28 19:16:53.336 16991 17068 I Unity   : Sentry: (Debug) Storing file /storage/emulated/0/Android/data/io.sentry.samples.unityofbugs/files/Sentry/618762E37120B6E01219C43B9076FFEC1ACE8406/1632871013_-444__-962146766.envelope.
09-28 19:16:53.336 16991 17026 I Unity   : Sentry: (Debug) Configuring the scope.
09-28 19:16:53.338 16991 17026 I Unity   : Sentry: (Debug) Android Scope Sync - Adding breadcrumb m:"Sentry: (Debug) Registering integration: 'UnityBeforeSceneLoadIntegration'. " l:"Info"
09-28 19:16:53.338 16991 17068 I Unity   : Sentry: (Debug) Envelope  handed off to transport. #1 in queue.
09-28 19:16:53.344 16991 17026 I Unity   : Sentry: (Debug) Registering integration: 'UnityBeforeSceneLoadIntegration'.
09-28 19:16:53.344 16991 17072 I Unity   : Sentry: (Debug) Worker signal triggered: flushing cached envelopes.
09-28 19:16:53.344 16991 17026 I Unity   : Sentry: (Debug) Configuring the scope.
09-28 19:16:53.344 16991 17026 I Unity   : Sentry: (Debug) Android Scope Sync - Adding breadcrumb m:"BeforeSceneLoad" l:"Info"
09-28 19:16:53.346 16991 17026 I Unity   : Sentry: (Debug) Registered BeforeSceneLoad integration.
09-28 19:16:53.347 16991 17026 I Unity   : Sentry: (Debug) Registering integration: 'SceneManagerIntegration'.
09-28 19:16:53.347 16991 17072 I Unity   : Sentry: (Debug) Reading cached envelope: /storage/emulated/0/Android/data/io.sentry.samples.unityofbugs/files/Sentry/618762E37120B6E01219C43B9076FFEC1ACE8406/__processing/1632871012_6820__1029138982.envelope
09-28 19:16:53.347 16991 17026 I Unity   : Sentry: (Debug) Registering SceneManager integration.
09-28 19:16:53.347 16991 17026 I Unity   : Sentry: (Debug) Registering integration: 'SessionIntegration'.
09-28 19:16:53.347 16991 17026 I Unity   : Sentry: (Debug) Registering Session integration.
09-28 19:16:53.364 16991 17026 I Unity   : Sentry: (Debug) Configuring the scope.
09-28 19:16:53.365 16991 17026 I Unity   : Sentry: (Debug) Android Scope Sync - Adding breadcrumb m:"Changed active scene to '1_BugfarmScene'" l:"Info"
09-28 19:16:53.365 16991 17072 I Unity   : Sentry: (Debug) Sending cached envelope:
09-28 19:16:53.367 16991 17026 I Unity   : Sentry: (Debug) Configuring the scope.
```

### PR DESCRIPTION
**Pull request by SimonCropp**
fixes #1218

### CODE REVIEW

File: test/Sentry.Tests/Internals/Http/CachingTransportTests.cs [hunk 1]
```diff
@@ -235,6 +235,37 @@ public async Task AwareOfExistingFiles()
...
+        using var envelope = Envelope.FromEvent(new SentryEvent());
+        await transport.SendEnvelopeAsync(envelope);
+
+        await transport.FlushAsync();
+
+        var message = _logger.Entries
+            .Where(x => x.Level == SentryLevel.Error)
+            .Select(x => x.RawMessage)
+            .Single();
+        Assert.Equal("Failed to send cached envelope: {0}, discarding cached envelope. Envelope contents: {1}", message);
```
Comment (lucas-zimerman) [inline test/Sentry.Tests/Internals/Http/CachingTransportTests.cs:266]:
```suggestion

        // Assert
        Assert.Equal("Failed to send cached envelope: {0}, discarding cached envelope. Envelope contents: {1}", message);
```

File: test/Sentry.Testing/TestOutputDiagnosticLogger.cs [hunk 2]
```diff
@@ -18,6 +18,8 @@ public class LogEntry
         public SentryLevel Level { get; set; }
         public string Message { get; set; }
         public Exception Exception { get; set; }
+        public object[] Arguments { get; set; }
```
Comment (lucas-zimerman) [inline test/Sentry.Testing/TestOutputDiagnosticLogger.cs:21]: Arguments got added but it's not used by any test?

**Review by lucas-zimerman — APPROVED** [review]
Added some comments but looks good for merging.

File: test/Sentry.Testing/TestOutputDiagnosticLogger.cs (hunk 2 above)
Comment (SimonCropp) [reply test/Sentry.Testing/TestOutputDiagnosticLogger.cs:21]: good catch.  will fix

**Pull request merged**